src/inspector.rs

```diff
@@ -0,0 +1,675 @@
+/// write, send, receive messages) and `0` when not, to indicate the
+/// impossibility of the further action, in which case, the inspector
+/// will stop.
+type OnWaitFrontendMessageOnPauseCallback =
+    dyn FnMut(*mut crate::v8_c_raw::bindings::v8_inspector_c_wrapper) -> std::os::raw::c_int;
+
+/// The debugging inspector, carefully wrapping the
+/// [`v8_inspector::Inspector`](https://chromium.googlesource.com/v8/v8/+/refs/heads/main/src/inspector)
+/// API.
+pub struct Inspector {
+    raw: *mut crate::v8_c_raw::bindings::v8_inspector_c_wrapper,
+    /// This callback is stored to preserve the lifetime, it is never
+    /// called by this object, but by the C++ side.
+    _on_response_callback: Option<Box<Box<OnResponseCallback>>>,
+    /// This callback is stored to preserve the lifetime, it is never
+    /// called by this object, but by the C++ side.
+    _on_wait_frontend_message_on_pause_callback:
+        Option<Box<Box<OnWaitFrontendMessageOnPauseCallback>>>,
+}
+
+impl std::fmt::Debug for Inspector {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let on_response_str = if let Some(ref callback) = self._on_response_callback {
+            format!("Some({callback:p})")
+        } else {
+            "None".to_owned()
+        };
+
+        let on_wait_str =
+            if let Some(ref callback) = self._on_wait_frontend_message_on_pause_callback {
+                format!("Some({callback:p})")
+            } else {
+                "None".to_owned()
+            };
+
+        f.debug_struct("Inspector")
+            .field("raw", &self.raw)
+            .field("_on_response_callback", &on_response_str)
+            .field("_on_wait_frontend_message_on_pause_callback", &on_wait_str)
+            .finish()
+    }
+}
+
+/// The callback function used to send the messages back to the client
+/// of the inspector (to the one who is debugging).
+extern "C" fn on_response(
+    string: *const ::std::os::raw::c_char,
+    rust_callback: *mut ::std::os::raw::c_void,
+) {
+    let string = unsafe { std::ffi::CStr::from_ptr(string) }.to_string_lossy();
+    log::trace!("Outgoing message: {string}");
+    let rust_callback: &mut Box<OnResponseCallback> = unsafe {
+        &mut *(rust_callback as *mut std::boxed::Box<dyn std::ops::FnMut(std::string::String)>)
+    };
+    rust_callback(string.to_string())
+}
+
+extern "C" fn on_wait_frontend_message_on_pause(
+    raw: *mut crate::v8_c_raw::bindings::v8_inspector_c_wrapper,
+    rust_callback: *mut ::std::os::raw::c_void,
+) -> ::std::os::raw::c_int {
+    log::trace!("on_wait_frontend_message_on_pause");
+    let rust_callback: &mut Box<OnWaitFrontendMessageOnPauseCallback> = unsafe {
+        &mut *(rust_callback
+            as *mut std::boxed::Box<
+                dyn std::ops::FnMut(*mut crate::v8_c_raw::bindings::v8_inspector_c_wrapper) -> i32,
+            >)
+    };
+    rust_callback(raw)
+}
+
+impl Inspector {
```

Comment:
I guess the inspector can not outlive the context? Should we try to express it somehow with lifetime?